src/Compilers/CSharp/Test/Emit2/Diagnostics/GetDiagnosticsTests.cs

```diff
@@ -734,6 +734,32 @@ void M2()
             Assert.True(eventQueue.IsCompleted);
         }
 
+        [Fact, WorkItem(67310, "https://github.com/dotnet/roslyn/issues/67310")]
+        public async Task TestArrowExpressionClauseAnalyzer()
+        {
+            var source = @"
+struct C
+{
+    int P => 0;
+    int this[int i] => 0;
+    int M() => 0;
```

Comment:
tests for constructors, destructors, operators, accessors (property, indexer, event plz).